typescript/src/ethereum.ts

```diff
@@ -562,3 +571,86 @@ class WalletRegistry extends EthereumContract {
     return publicKey.substring(2)
   }
 }
+
+/**
+ * Implementation of the Ethereum TBTCVault handle.
+ */
+export class TBTCVault extends EthereumContract implements ChainTBTCVault {
+  constructor(config: ContractConfig) {
+    super(config, TBTCVaultDeployment)
+  }
+
+  // eslint-disable-next-line valid-jsdoc
+  /**
+   * @see {ChainTBTCVault#requestOptimisticMint}
+   */
+  async requestOptimisticMint(
+    depositTxHash: TransactionHash,
+    depositOutputIndex: number
+  ): Promise<void> {
+    await this._instance.requestOptimisticMint(
+      depositTxHash,
+      depositOutputIndex
+    )
+  }
+
+  // eslint-disable-next-line valid-jsdoc
+  /**
+   * @see {ChainTBTCVault#optimisticMintingRequests}
+   */
+  async optimisticMintingRequests(
+    depositTxHash: TransactionHash,
+    depositOutputIndex: number
+  ): Promise<OptimisticMintingRequest> {
+    const depositKey = Bridge.buildDepositKey(depositTxHash, depositOutputIndex)
+
+    const request = await this._instance.optimisticMintingRequests(depositKey)
+
+    return this.parseOptimisticMintingRequest(request)
+  }
+
+  /**
+   * Parses a optimistic minting request using data fetched from the on-chain contract.
+   * @param request Data of the optimistic minting request.
+   * @returns Parsed optimistic minting request.
+   */
+  private parseOptimisticMintingRequest(
+    request: any
+  ): OptimisticMintingRequest {
+    return {
+      requestedAt: BigNumber.from(request.requestedAt).toNumber(),
+      finalizedAt: BigNumber.from(request.finalizedAt).toNumber(),
+    }
+  }
+
+  // TODO: Add docs
+  // TODO: Consider exposing in optimistic-minting.ts and in chain.ts
```

Comment:
Just a note that we need to make it in the same way as agreed during [this discussion](https://github.com/keep-network/tbtc-v2/pull/457#discussion_r1062288531)